[ews-build] Upload jscore-test logs to S3

briannafan opened the pull request:
#### c5f4de47e18798066c9f5cbd73ef67701ae1562b
<pre>
[ews-build] Upload jscore-test logs to S3
<a href="https://bugs.webkit.org/show_bug.cgi?id=270649">https://bugs.webkit.org/show_bug.cgi?id=270649</a>
<a href="https://rdar.apple.com/124223332">rdar://124223332</a>

Reviewed by Aakash Jain.

Filters JSC logs into a log file and uploads it to S3.

* Tools/CISupport/ews-build/steps.py:
(RunJavaScriptCoreTests):
(RunJavaScriptCoreTests.start):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/278525@main">https://commits.webkit.org/278525@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cd3cefd2acd51eb5300ef1d8c659d170dc66863b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50856 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30153 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3174 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54115 "Built successfully") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1547 "Failed to checkout and rebase branch from PR 28107") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53159 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36429 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1198 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/41428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52955 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27796 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43809 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/22556 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/25139 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/1065 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9297 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/1127 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55709 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/25958 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/1012 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/48837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/50877 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/27215 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43884 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/47920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/28083 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7364 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/26947 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3611,6 +3611,22 @@ def start(self):
         self.setCommand(self.command + customBuildFlag(self.getProperty('platform'), self.getProperty('fullPlatform')))
         self.command.extend(self.command_extra)
+
+        self.command = self.shell_command(' '.join(self.command) + ' 2>&1 | python3 Tools/Scripts/filter-jsc-tests.py')
```
Why isn't this script `filter-jsc-tests`?

We also need to cherry-pick this filter script to a few branches before deploying this.

aj062 replied:
Good point about the branch. Let's do that and land this soon (maybe with little bit delay). 
Happy to see this PR :)

aj062 reviewed (approved):
r=me, assuming it's tested.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3585,6 +3585,7 @@ class RunJavaScriptCoreTests(shell.Test, AddToLogMixin):
 
     def __init__(self, **kwargs):
         super().__init__(logEnviron=False, sigtermTime=10, **kwargs)
+        kwargs['timeout'] = 2 * 60 * 60
```
This timeout doesn't work, since kwargs is not used after setting it here. It should be done before the super() call. 
However, let's just maintain consistency with rest of this file and pass timeout as a function parameter to super().__init__().

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -3611,6 +3611,22 @@ def start(self):
+                content_type='text/plain',
+            )
+        ]
+        self.build.addStepsAfterCurrentStep(steps_to_add)
```
I tested it on UAT instance and found this this doesn't really work.
This is because of code in evaluateCommand() method in this case.

In case of success we call self.build.buildFinished() which finish the build immediately without calling the steps for uploading the build logs.
In case of failure, it does another self.build.addStepsAfterCurrentStep() call which calls those steps before these steps. (see https://github.com/buildbot/buildbot/issues/4874).

Better to move this code inside evaluateCommand() method and ensure there is single call to self.build.addStepsAfterCurrentStep.

Note that one difference in this class as compared to build.webkit.org is that from build.webkit.org uses new-style run method while this code uses old-style start() method.

briannafan replied:
Thank you for explaining! Made the fix.

aj062 reviewed (approved):
r+ assuming it's tested. Please have a look at unit-tests as well.

briannafan committed:
[ews-build] Upload jscore-test logs to S3
https://bugs.webkit.org/show_bug.cgi?id=270649
rdar://124223332

Reviewed by Aakash Jain.

Filters JSC logs into a log file and uploads it to S3.

* Tools/CISupport/ews-build/steps.py:
(RunJavaScriptCoreTests):
(RunJavaScriptCoreTests.start):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/278525@main
